_P208_Nokia_LCD_5110.ino

```diff
@@ -0,0 +1,290 @@
+        Settings.TaskDevicePluginConfig[event->TaskIndex][2] = getFormItemInt(F("plugin_208_rotation"));
+        Settings.TaskDevicePluginConfig[event->TaskIndex][3] = getFormItemInt(F("plugin_208_charsize_line_1"));
+        Settings.TaskDevicePluginConfig[event->TaskIndex][4] = getFormItemInt(F("plugin_208_charsize_line_2"));
+        Settings.TaskDevicePluginConfig[event->TaskIndex][5] = getFormItemInt(F("plugin_208_charsize_line_3"));
+
+        char deviceTemplate[lcd_lines][digits_per_line];
+        for (byte varNr = 0; varNr < lcd_lines; varNr++)
+        {
+          char argc[25];
+          String arg = F("Plugin_208_template");
+          arg += varNr + 1;
+          arg.toCharArray(argc, 25);
+          String tmpString = web_server.arg(argc);
+          strncpy(deviceTemplate[varNr], tmpString.c_str(), sizeof(deviceTemplate[varNr]));
+        }
+
+      //  Settings.TaskDeviceID[event->TaskIndex] = 1; // temp fix, needs a dummy value
+
+        SaveCustomTaskSettings(event->TaskIndex, (byte*)&deviceTemplate, sizeof(deviceTemplate));
+        success = true;
+        break;
+      }
+
+    case PLUGIN_INIT:{
+        if (!lcd3)
+        {
+         // #define RST  4 //LCD_pin_1 reset
+         // #define CE   5 //LCD_pin_2 chip select
+         // #define DC  12 //LCD_pin_3 Data/Command select
+         // #define DIN 13 //LCD_pin_4 Serial data 
+         // #define CLK 14 //LCD_pin_5 Serial clock out
+         //lcd3 = new Adafruit_PCD8544(CLK, DIN, DC, CE, RST);
+
+          #if defined(ESP8266)
+              // SPI interface
+              lcd3 = new Adafruit_PCD8544(15, 2, -1); // D/C (need to be connected) GPIO2 , CS(CEN) GPIO15 , RESET not connected
+          #endif
+          #if defined(ESP32)
+              // SPI interface:
+              //#define RST  - //LCD_pin_1 reset connected to Vcc with 10k resistor
+                #define CE   5 //LCD_pin_2 chip select
+                #define DC  12 //LCD_pin_3 Data/Command select
+              //#define DIN 23 //LCD_pin_4 Serial data 
+              //#define CLK 18 //LCD_pin_5 Serial clock out
+              // In hardware tab; 
+              // - enable SPI; 
+              // - Select VSPI:CLK=GPIO-18, MISO=GPIO-19, MOSI=GPIO-23 
+              //   miso not used 
+               lcd3 = new Adafruit_PCD8544(DC, CE, -1); 
+          #endif
+        }
+        // Setup lcd3 display
+        byte plugin1 = Settings.TaskDevicePluginConfig[event->TaskIndex][2]; // rotation
+        byte plugin2 = Settings.TaskDevicePluginConfig[event->TaskIndex][1]; // contrast
+        byte plugin4 = Settings.TaskDevicePluginConfig[event->TaskIndex][0]; // backlight_onoff
+        UserVar[event->BaseVarIndex+2]=plugin1;
+        UserVar[event->BaseVarIndex+1]=plugin2;
+        UserVar[event->BaseVarIndex]=!plugin4;
+        lcd3->begin();
+        lcd3->setContrast(30);
+        lcd3->setContrast(plugin2);
+        lcd3->setRotation(plugin1);
+ 
+        delay(1000);

```

Comment:
Such long delay calls are a big no-no!
If you need to do things using a delay like this, please keep track of things still to-do and perform them in a later call (e.g. PLUGIN_ONCE_A_SECOND)

Reply:
No Idea why there is a 1 second delay. I removed this line without consequence at testing.